zowe/zowe-cli
# [zosfiles] Add stream support to Download API and refactor Get API

**What It Does**
Add stream support to APIs for downloading data sets and USS files, similar to what was done for job spool files in https://github.com/zowe/zowe-cli/pull/1686.

Also refactor the Get APIs to call the Download API so they take advantage of all its features including USS file tag detection.

**How to Test**
Verify that the Get APIs work as expected with a test script like the following:
```javascript
const zowe = require("@zowe/cli");

(async () => {
    // Load connection info from default z/OSMF profile
    const profInfo = new zowe.imperative.ProfileInfo("zowe");
    await profInfo.readProfilesFromDisk();
    const zosmfProfAttrs = profInfo.getDefaultProfile("zosmf");
    const zosmfMergedArgs = profInfo.mergeArgsForProfile(zosmfProfAttrs, { getSecureVals: true });
    const session = zowe.imperative.ProfileInfo.createSession(zosmfMergedArgs.knownArgs);

    // Test Zowe SDKs
    const dsBuffer = await zowe.Get.dataSet(session, "<dsname>");
    console.log(dsBuffer.toString());
    const ussBuffer = await zowe.Get.USSFile(session, "<ussname>");
    console.log(ussBuffer.toString());
})();
```

You can also perform cross-LPAR copy using streams without downloading the file to disk as an intermediate step:
```javascript
const responseStream = new require("stream").PassThrough();
zowe.Download.ussFile(sourceSession, "<sourceFilePath>", { binary: true, stream: responseStream });
await zowe.Upload.streamToUssFile(targetSession, "<targetFilePath>", responseStream, { binary: true });
```

**Review Checklist**
I certify that I have:


**Additional Comments**
